### DISCUSSION TITLE
Error with lc_geometry model i think

### DISCUSSION THREAD

**Discussion by davidandres8** [phoebe-project/phoebe2#743] [Questions]
Hi, I am trying to fit a binary curve through the ebai_mlp, lc_geometry and nelder mead models, there is no problem when graphing the light curve, but it returns an error in the lc_geometry method part:

I attach the error lines

2G model C failed with error: Residuals are not finite in the initial point.
2G model CE failed with error: Residuals are not finite in the initial point.
2G model CG failed with error: Residuals are not finite in the initial point.
2G model CGE failed with error: Residuals are not finite in the initial point.
2G model CG12 failed with error: Residuals are not finite in the initial point.
2G model CG12E1 failed with error: Residuals are not finite in the initial point.
2G model CG12E2 failed with error: Residuals are not finite in the initial point.

TypeError: TwoGaussianModel.ce() missing 2 required positional arguments: 'Aell' and 'phi0'

Greetings!

**Comment by kecnry** [comment]
I'm not sure what's going on here - hopefully these questions can help us track down the problem:

* Do you have uncertainties attached to your data?
* Are there any nans or non-finite values in the fluxes or uncertainties?
* Can you attach the full traceback and/or a minimal script that can reproduce this error message?

**Comment by davidandres8** [comment]
Hi, I attached a screenshot of the error.

Greetings!

El mié, 7 jun 2023 a las 8:51, Kyle Conroy ***@***.***>)
escribió:

> I'm not sure what's going on here - hopefully these questions can help us
> track down the problem:
>
>    - Do you have uncertainties attached to your data?
>    - Are there any nans or non-finite values in the fluxes or
>    uncertainties?
>    - Can you attach the full traceback and/or a minimal script that can
>    reproduce this error message?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/phoebe-project/phoebe2/discussions/743#discussioncomment-6111235>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANNVRIXW4KJXBFGN5Y7YR6TXKB2N3ANCNFSM6AAAAAAYXQQJ6Y>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Se despide cordialmente,
David Andrés Fernández Pérez
Licenciado en Astronomía
Universidad de La Serena, Chile

**Reply by kecnry** [reply]
It looks like the attachment was dropped.  Can you please copy and paste the text into a text document and attach it on github or paste it into the message surrounding by triple ticks "```"?  And if you're unable to share the data or script publicly, please feel free to email that to me directly so that I can reproduce the issue and see what is happening.

**Reply by davidandres8** [reply]
https://mail.google.com/mail/u/0?ui=2&ik=15c80ca26a&attid=0.1&permmsgid=msg-a:r-2504539226346018867&th=1889c2a4136d1439&view=att&disp=safe&realattid=f_ling1fss0
maybe now you can see the picture?

**Reply by kecnry** [reply]
No I cannot, please attach it directly to GitHub instead of through your mail client or as an external link.  And text will be much easier to parse and search than a screenshot if you still have the full traceback in text form.

**Reply by davidandres8** [reply]
`try:

    print(b.filter(solver='ebai_mlp'))

    b.run_solver('ebai_mlp', solution='ebai_mlp_solution')

    afig, mplfig = b.plot(solution='ebai_sol', show=True)

except Exception as err:

    print(err)

ParameterSet: 7 parameters
         comments@ebai_mlp@solver: 
       use_server@ebai_mlp@solver: none
      lc_datasets@ebai_mlp@solver: ['*']
        phase_bin@ebai_mlp@solver: True
      phase_nbins@ebai_mlp@solver: 500
      ebai_method@ebai_mlp@solver: mlp
            orbit@ebai_mlp@solver: binary
2G model C failed with error: Residuals are not finite in the initial point.
2G model CE failed with error: Residuals are not finite in the initial point.
2G model CG failed with error: Residuals are not finite in the initial point.
2G model CGE failed with error: Residuals are not finite in the initial point.
2G model CG12 failed with error: Residuals are not finite in the initial point.
2G model CG12E1 failed with error: Residuals are not finite in the initial point.
2G model CG12E2 failed with error: Residuals are not finite in the initial point.
TwoGaussianModel.ce() missing 2 required positional arguments: 'Aell' and 'phi0'
`

**Reply by kecnry** [reply]
can you remove the try/except so that the full traceback is shown?  It should point to which lines of code are raising the errors which will help to see the actual cause of the problem.  Can you also either send your data, bundle, or a minimial script to reproduce what you're seeing?  Without that, we cannot diagnose the problem or fix it since this does not happen in all cases.

**Reply by davidandres8** [reply]
`print(b.filter(solver='ebai_mlp'))

b.run_solver('ebai_mlp', solution='ebai_mlp_solution')

afig, mplfig = b.plot(solution='ebai_sol', show=True)

ParameterSet: 7 parameters
         comments@ebai_mlp@solver: 
       use_server@ebai_mlp@solver: none
      lc_datasets@ebai_mlp@solver: ['*']
        phase_bin@ebai_mlp@solver: True
      phase_nbins@ebai_mlp@solver: 500
      ebai_method@ebai_mlp@solver: mlp
            orbit@ebai_mlp@solver: binary
2G model C failed with error: Residuals are not finite in the initial point.
2G model CE failed with error: Residuals are not finite in the initial point.
2G model CG failed with error: Residuals are not finite in the initial point.
2G model CGE failed with error: Residuals are not finite in the initial point.
2G model CG12 failed with error: Residuals are not finite in the initial point.
2G model CG12E1 failed with error: Residuals are not finite in the initial point.
2G model CG12E2 failed with error: Residuals are not finite in the initial point.

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In [6], line 2
      1 print(b.filter(solver='ebai_mlp'))
----> 2 b.run_solver('ebai_mlp', solution='ebai_mlp_solution')
      3 afig, mplfig = b.plot(solution='ebai_sol', show=True)

File ~/.local/lib/python3.10/site-packages/phoebe/parameters/parameters.py:424, in send_if_client.<locals>._send_if_client(self, *args, **kwargs)
    421     return ret_
    423 else:
--> 424     return fctn(self, *args, **kwargs)

File ~/.local/lib/python3.10/site-packages/phoebe/frontend/bundle.py:13649, in Bundle.run_solver(self, solver, solution, detach, return_changes, **kwargs)
  13645     return job_param
  13648 solver_class = getattr(_solverbackends, '{}Backend'.format(solver_ps.kind.title()))
> 13649 params = solver_class().run(self, solver_ps.solver, compute, solution=solution, **{k:v for k,v in kwargs.items() if k not in ['compute']})
  13650 metawargs = {'context': 'solution',
  13651              'solver': solver_ps.solver,
  13652              'compute': compute,
  13653              'kind': solver_ps.kind,
  13654              'solution': solution}
  13657 comment_param = StringParameter(qualifier='comments', value=kwargs.get('comments', solver_ps.get_value(qualifier='comments', default='', **_skip_filter_checks)), description='User-provided comments for this solution.  Feel free to place any notes here.')

File ~/.local/lib/python3.10/site-packages/phoebe/solverbackends/solverbackends.py:555, in BaseSolverBackend.run(self, b, solver, compute, **kwargs)
    552     rpacketlists_per_worker = [self.run_worker(**packet)]
    554 else:
--> 555     rpacketlists_per_worker = [self.run_worker(**packet)]
    557 logger.debug("rank:{}/{} calling _fill_solution".format(mpi.myrank, mpi.nprocs))
    558 return self._fill_solution(solution_ps, rpacketlists_per_worker)

File ~/.local/lib/python3.10/site-packages/phoebe/solverbackends/solverbackends.py:1250, in EbaiBackend.run_worker(self, b, solver, compute, **kwargs)
   1247 else:
   1248     lcModel = ligeor.models.Polyfit(phases=phases_shifted[s], fluxes=fluxes[s], sigmas=sigmas[s])
-> 1250 lcModel.fit()
   1251 ebai_phases = np.linspace(-0.5,0.5,ebai_phase_bins)
   1252 ebai_fluxes = lcModel.compute_model(ebai_phases, best_fit=True)

File ~/.local/lib/python3.10/site-packages/phoebe/dependencies/ligeor/models/twogaussian.py:102, in TwoGaussianModel.fit(self, fit_funcs, param_vals)
     99     self.fit_twoGaussian_models()
    100     # compute all model light curves
--> 102 self.compute_twoGaussian_models()
    103 # compute corresponding BIC values
    104 self.compute_twoGaussian_models_BIC()

File ~/.local/lib/python3.10/site-packages/phoebe/dependencies/ligeor/models/twogaussian.py:621, in TwoGaussianModel.compute_twoGaussian_models(self)
    618 models = {}
    620 for fkey in self.fits.keys():
--> 621     models[fkey] = self.twogfuncs[fkey](self.phases, *self.fits[fkey])
    623 self.models = models

TypeError: TwoGaussianModel.ce() missing 2 required positional arguments: 'Aell' and 'phi0'

`

**Reply by davidandres8** [reply]
maybe I can send you the complete code by mail?

**Reply by kecnry** [reply]
sure - you should be able to find my email on my GitHub profile.  These easiest way so that the data are also included would be to save and send the bundle `b.save('before_run_solver.bundle')` immediately before the `run_solver` line that fails.  That way you don't need to send multiple files and we don't need to re-run any previous steps in the script.

**Reply by kecnry** [reply]
it looks like one of the bins is empty resulting in this error.  You can get around that by lowering the number of bins (`phase_nbins` in the solver options - 100 seemed to work successfully on your data for me) or by disabling binning entirely (`phase_bin = False`).  

I also created #749 to try to either fix this automatically internally or at least raise an error that suggests modifying the binning options.

**Comment by davidandres8** [comment]
How can I know what is the right number of bins for each light curve? since
I must make the adjustment for about 8 curves.

Greetings!

El El sáb, 10 de jun. de 2023 a la(s) 14:19, Kyle Conroy <
***@***.***> escribió:

> it looks like one of the bins is empty resulting in this error. You can
> get around that by lowering the number of bins (phase_nbins in the solver
> options - 100 seemed to work successfully on your data for me).
>
> I also created #749 <https://github.com/phoebe-project/phoebe2/issues/749>
> to try to either fix this automatically internally or at least raise an
> error that suggests modifying the binning options.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/phoebe-project/phoebe2/discussions/743#discussioncomment-6141196>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANNVRIQMKFVSWL4GDJDNW2TXKS3CPANCNFSM6AAAAAAYXQQJ6Y>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 
Se despide cordialmente,
David Andrés Fernández Pérez
Licenciado en Astronomía
Universidad de La Serena, Chile

**Reply by kecnry** [reply]
You can either bin manually outside phoebe and ensure that there are no empty bins or you can disable binning entirely with `phase_bin=False`.

**Comment by davidandres8** [comment]
HI Kyle,
I followed the steps in the tutorials and the adjustment with ebai_mlp
looks like this, what could I do to improve the adjustment?

Greetings

El sáb, 10 jun 2023 a las 14:23, Kyle Conroy ***@***.***>)
escribió:

> You can either bin manually outside phoebe and ensure that there are no
> empty bins or you can disable binning entirely with phase_bin=False.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/phoebe-project/phoebe2/discussions/743#discussioncomment-6141209>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANNVRIWAF6X5PCN2SMWTRDTXKS3QZANCNFSM6AAAAAAYXQQJ6Y>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Se despide cordialmente,
David Andrés Fernández Pérez
Licenciado en Astronomía
Universidad de La Serena, Chile
